src/quagga.js

```diff
@@ -195,14 +201,29 @@ function transformResult(result) {
     }
 }
 
+function addResult (result, imageData) {
+    var i;
+
+    if (!imageData || !result || !_resultCollector) {
+        return;
+    }
+
+    if (result.barcodes) {
+        for (i = 0; i < result.barcodes.length; i++) {
+            addResult(result.barcodes[i], imageData);
+        }
+        return;
+    }
+
+    if (result.codeResult) {
+        _resultCollector.addResult(imageData, _inputStream.getCanvasSize(), result.codeResult);
+    }
+}
+
 function publishResult(result, imageData) {
     if (_onUIThread) {
         transformResult(result);
-        if (imageData && result && result.codeResult) {
-            if (_resultCollector) {
-                _resultCollector.addResult(imageData, _inputStream.getCanvasSize(), result.codeResult);
-            }
-        }
+        addResult(result, imageData);
     }
 
     Events.publish("processed", result);
```

Comment:
I must have overlooked the last part of that method the first time through, good catch.  I just pushed a fix for it which also future-proofs the check if we decide to return an array instead of an object.  I'm making the assumption that some of the barcodes might not have a `codeResult` instead of just checking `barcodes.length`.  If you agree that's the way we should go I will commit the array result change.